### ISSUE TITLE
exit code handling between SysV-compat and Systemd service is inconsistent

### ISSUE THREAD

**Issue by RangerRick** [systemd/systemd#12580]
**systemd version the issue has been seen with**
> 237, but the code in github shows `master` [has the same behavior](https://github.com/systemd/systemd/blob/master/src/sysv-generator/sysv-generator.c#L178).

**Used distribution**
 > Ubuntu

**Expected behaviour you didn't see**
 > `systemctl start <service>` should exit non-0 if the service is a SysV script and returns non-0.

**Unexpected behaviour you saw**
 > `systemctl start <service>` exits 0.

**Steps to reproduce the problem**

1. Create an `/etc/init.d/exitcode` script with the following contents:
```
 #! /bin/sh

### BEGIN INIT INFO
# Provides: exitcode
# Required-Start: $local_fs
# Default-Start: 2 3 4 5
# Default-Stop: 0 1 6
# Short-Description: Test Exit Code (SysV)
# Description: Test case for Systemd SysV compatibility layer
### END INIT INFO

exit 6
```
2. Run `systemctl enable exitcode`
3. Run `systemctl start exitcode`

Output:
```
ranger@carolina:/etc/init.d$ sudo systemctl enable exitcode
exitcode.service is not a native service, redirecting to systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable exitcode
ranger@carolina:/etc/init.d$ sudo systemctl start exitcode
ranger@carolina:/etc/init.d$ echo $?
0
```

It should exit non-0 because an exit code of 6 is > 0, even though it represents "not configured". 

Exiting non-0 is the default behavior of native Systemd `.service` files that exit with code 6:

```
 [Unit]
Description=Test Exit Code (Systemd)

[Service]
Type=oneshot
ExecStart=/bin/bash -c 'exit 6'

[Install]
WantedBy=default.target
```

Output:
```
ranger@carolina:/lib/systemd/system$ sudo systemctl start exitcode
Job for exitcode.service failed because the control process exited with error code.
See "systemctl status exitcode.service" and "journalctl -xe" for details.
ranger@carolina:/lib/systemd/system$ echo $?
1
```

**TL;DR**

If Systemd is going to treat "not configured" as a non-failure in SysV compatibility mode, it should do so in native services too. Otherwise, it should error in both.  Traditionally, non-0 has _always_ been an error, and the additional LSB codes are just hints as to _why_ it's an error, so the latter would provide better compatibility with expectations, _and_ with Systemd service files.

**Comment by poettering** [comment]
well, there are plenty of LSB services that exit 5 or 6 when they are not configured and expect this not to be a failure case, i.e. expect this not to result in an all red boot screen. That's why generated native services for LSB services contain the `SuccessExitStatus=5 6` line, to make sure this is accepted. 

Native services don't follow semantics like that, we expect that people use ConditionXYZ= or so, or exit cleanly on failure, or configure their own SuccessExitStatus= as they like. And precisely that is what I'd recommend you too here: if you want behaviour like this, by all means add `SuccessExitStatus=5 6` to your service file and you get exactly the behaviour you want.

Closing, since this is already supported. Hope that makes sense.

**Comment by RangerRick** [comment]
Not being fatal for startup during boot I can at least understand... but does it not seem wrong for `systemctl start <service>` to _somehow_ indicate that it wasn't perfect success?  A return code of `0` tells the user "your service should now be running" but it won't be.
